brainiak/matnormal/matnormal_likelihoods.py

```diff
@@ -64,147 +100,214 @@ def solve_det_conditional(x, sigma, A, Q):
+    rowsize = tf.cast(tf.shape(input=x)[0], "float64")
+    colsize = tf.cast(tf.shape(input=x)[1], "float64")
 
-    solve_col = col_cov.Sigma_inv_x(tf.transpose(x))
+    solve_col = col_cov.solve(tf.transpose(a=x))
     logdet_col = col_cov.logdet
 
-    solve_row, logdet_row = solve_det_marginal(x, row_cov, marg,
-                                               marg_cov)
+    solve_row, logdet_row = solve_det_marginal(x, row_cov, marg, marg_cov)
 
-    return _mnorm_logp_internal(colsize, rowsize, logdet_row,
-                                logdet_col, solve_row, solve_col)
+    return _mnorm_logp_internal(
+        colsize, rowsize, logdet_row, logdet_col, solve_row, solve_col
+    )
 
 
 def matnorm_logp_marginal_col(x, row_cov, col_cov, marg, marg_cov):
     """
-    Log likelihood for centered matrix-variate normal density. Assumes that
-    row_cov, col_cov, and marg_cov follow the API defined in CovBase.
+    Log likelihood for centered marginal matrix-variate normal density.
 
-    When you marginalize in mnorm, you end up with a covariance S + APA',
-    where P is the covariance of A in the relevant dimension.
+    .. math::
+        X \\sim \\mathcal{MN}(0, R, Q)\\
+        Y \\mid \\X \\sim \\mathcal{MN}(XA, R, C),\\
+        Y \\sim \\mathcal{MN}(0, R, C + AQA)
+
+    This function efficiently computes the marginals by unpacking some
+    info in the covariance classes and then dispatching to solve_det_marginal.
+
+    Parameters
+    ---------------
+    x: tf.Tensor
+        Observation tensor
+    row_cov: CovBase
+        Row covariance implementing the CovBase API
+    col_cov: CovBase
+        Column Covariance implementing the CovBase API
+    marg: tf.Tensor
+        Marginal factor
+    marg_cov: CovBase
+        Prior covariance implementing the CovBase API
 
-    This method exploits the matrix inversion and determinant lemmas to
-    construct S + APA' given the covariance API in in CovBase.
     """
-    rowsize = tf.cast(tf.shape(x)[0], 'float64')
-    colsize = tf.cast(tf.shape(x)[1], 'float64')
+    rowsize = tf.cast(tf.shape(input=x)[0], "float64")
+    colsize = tf.cast(tf.shape(input=x)[1], "float64")
 
-    solve_row = row_cov.Sigma_inv_x(x)
+    solve_row = row_cov.solve(x)
     logdet_row = row_cov.logdet
 
-    solve_col, logdet_col = solve_det_marginal(tf.transpose(x),
-                                               col_cov,
-                                               tf.transpose(marg),
-                                               marg_cov)
+    solve_col, logdet_col = solve_det_marginal(
+        tf.transpose(a=x), col_cov, tf.transpose(a=marg), marg_cov
+    )
 
-    return _mnorm_logp_internal(colsize, rowsize, logdet_row,
-                                logdet_col, solve_row, solve_col)
+    return _mnorm_logp_internal(
+        colsize, rowsize, logdet_row, logdet_col, solve_row, solve_col
+    )
 
 
 def matnorm_logp_conditional_row(x, row_cov, col_cov, cond, cond_cov):
     """
```

Comment:
I wonder if you would consider adding docstring for this function

Reply:
added